code/WorkInProgress/JobXPRewards.dm

```diff
@@ -225,10 +225,17 @@ mob/verb/checkrewards()
 
 	activate(var/client/C)
 		var/found = 0
-
 		var/O = locate(sacrifice_path) in C.mob.contents
 		if (istype(O, sacrifice_path))
 			var/obj/item/gun/energy/egun/K = O
+			if (K.nojobreward) // Checks to see if it was scanned by a device analyzer
+				boutput(C.mob, "This [sacrifice_name] has forever been ruined by a device analyzer's magnets. It can't turn into a sword ever again!!")
+				src.claimedNumbers[usr.key] --
+				return
+			if (K.deconstruct_flags == DECON_BUILT) //Checks to see if it was built from a frame
```

Comment:
Energy guns probably won't ever have any deconstruction flags besides `DECON_BUILT`, but even then, probably use `if (K.deconstruct_flags & DECON_BUILT)`.

Reply:
did it!!! was also suggested to me by pali